tests/loops/test_loop_state_dict.py

```diff
@@ -40,15 +40,209 @@ def test_loops_state_dict_structure():
         "test_loop": trainer.test_loop.state_dict(),
         "predict_loop": trainer.predict_loop.state_dict(),
     }
+    # todo (tchaton) Update this once new progress as been added.
+    # yapf: disable
     expected = {
         "fit_loop": {
-            'epoch_loop': {
-                'batch_loop': {},
-                'val_loop': {},
+            "epoch_loop": {
+                "batch_loop": {
+                    "state_dict": {},
+                    "progress": {
```

Comment:
The progress state dicts shoild be replaced with `ANY` as we already test them in

https://github.com/PyTorchLightning/pytorch-lightning/blob/370fa67004a0132733df91976490b7ed284bc406/tests/trainer/test_progress.py#L138

Otherwise the tests should be merged

Reply:
I prefer to explicit show the structure of the `state dict`. This is easier for people to understand its true structure.